model/task/task.go

```diff
@@ -2472,12 +2472,24 @@ func (t *Task) Insert() error {
 // are also archived.
 func (t *Task) Archive() error {
 	if t.DisplayOnly && len(t.ExecutionTasks) > 0 {
-		execTasks, err := FindAll(db.Query(ByIds(t.ExecutionTasks)))
-		if err != nil {
-			return errors.Wrap(err, "retrieving execution tasks")
-		}
-		if err = ArchiveMany(execTasks); err != nil {
-			return errors.Wrap(err, "archiving execution tasks")
+		if !t.ResetFailedWhenFinished {
+			execTasks, err := FindAll(db.Query(ByIds(t.ExecutionTasks)))
+
+			if err != nil {
+				return errors.Wrap(err, "retrieving execution tasks")
+			}
+			if err = ArchiveMany(execTasks, t.Execution); err != nil {
```

Comment:
Would be nice to call this outside of the if/else cases instead and just assign Find() to a neutral execTasksToArchive or something (will need to create the variable outside of the if/else statements to use the variable outside also)

Reply:
The error messages are slightly different to different, would you recommend merging the error messages to the same to do this?